src/app/ReviewService.Abstractions/ReviewConditionDelegate.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ReviewService.Abstractions;
+
+/// <summary>
+///  A condition used to determine if a review should be requested based on <see cref="ReviewSettings"/>.
+/// </summary>
+/// <param name="ct"><see cref="CancellationToken"/>.</param>
+/// <param name="reviewPromptSettings">The persisted <see cref="ReviewSettings"/> tracked by the application.</param>
+/// <param name="currentDateTime">The current date and time.</param>
+/// <returns>If a review should be requested.</returns>
+public delegate Task<bool> ReviewConditionDelegate(CancellationToken ct, ReviewSettings reviewPromptSettings, DateTimeOffset currentDateTime);
```

Comment:
You probably want this delegate to be generic instead of directly using `ReviewSettings`.

Reply:
Nice catch! 👀